packages/lit-html/src/directive-helpers.ts

```diff
@@ -16,22 +16,32 @@ import {
   AttributePart,
   NodePart,
   Part,
-  nothing,
   DirectiveParent,
+  TemplateResult,
+  DirectiveResult,
 } from './lit-html.js';
 
+type Primitive = null | undefined | boolean | number | string | symbol | bigint;
+
 /**
- * The state of a NodePart, which can be detached and reattached.
+ * Tests if a value is a primitive value.
+ *
+ * See https://tc39.github.io/ecma262/#sec-typeof-operator
  */
-export type NodePartState = {};
+export const isPrimitive = (value: unknown): value is Primitive =>
+  value === null || (typeof value != 'object' && typeof value != 'function');
 
 /**
- * The private interface for NodePartState, which should be kept opaque.
+ * Tests if a value is a TemplateResult.
  */
-type NodePartStateInternal = {
-  _value: unknown;
-  _fragment: DocumentFragment;
-};
+export const isTemplateResult = (value: unknown): value is TemplateResult =>
+  (value as TemplateResult)._$litType$ !== undefined;
```

Comment:
```suggestion
  (value as TemplateResult)?._$litType$ !== undefined;
```
So it's safe to use on any value? (same below for `isDirectiveResult`)

Reply:
done